pkg/odo/cli/logs/logs.go

```diff
@@ -133,17 +135,44 @@ func (o *LogsOptions) Run(ctx context.Context) error {
 	}
 
 	uniqueContainerNames := map[string]struct{}{}
+	errChan := make(chan error)
+	wg := sync.WaitGroup{}
+	var mu sync.Mutex
 	for _, entry := range containersLogs {
 		for container, logs := range entry {
 			uniqueName := getUniqueContainerName(container, uniqueContainerNames)
 			uniqueContainerNames[uniqueName] = struct{}{}
-			err = printLogs(uniqueName, logs, o.out)
-			if err != nil {
-				return err
-			}
+			l := logs
+			wg.Add(1)
+			go func(out io.Writer, follow bool) {
```

Comment:
Without the `--follow` flag, the logs will be displayed in a random order (the one acquiring the lock first, etc).

IMO, you could run the function as non go routine for non-follow, and run it as go routine for follow, so you don't need the use of mutex and you respect the order of the logs.

Reply:
Yes. That's something that crossed my mind as well when I noticed the ordering. But I didn't make the change to be able to reuse the same implementation. Anyway, I'll make the change.